README.md

```diff
@@ -1 +1,184 @@
-# json-manipulation-lang
\ No newline at end of file
+# JML (JSON Manipulation Language) Overview
+
+JML (JSON Manipulation Language) is a purely functional domain-specific language designed for seamless manipulation and transformation of JSON data. With a focus on immutability and function composition, JML allows users to build complex JSON structures and perform operations using functions, lambdas, and other functional constructs.
+
+## Basic Syntax
+
+### Variables
+In JML, variables are immutable once assigned. You can bind values to names:
+
+```jml
+a = 1
+d = "some_string"
+```
+
+### JSON Object Construction
+Embed variable bindings directly into JSON objects. The code is divided into two sections: the header, which contains all variable bindings (statements), and the body, separated by the '---' keyword, containing a single expression: 
+
+**Header:**
+```jml
+a = 1
+d = "some_string"
+c = \x y. x + y
+```
+
+**Body:**
+```jml
+{
+    "key1": a, 
+    "key2": d,
+    "key3": c(5, 3)
+}
+```
+
+### Lambdas
+JML supports lambda expressions, which are anonymous functions. The syntax for lambdas is `\x y. expression`, where `x` and `y` are parameters:
```

Comment:
Yes, you got it almost right. There are no functions only variables, but you can bind lambda expression with the name so that you can operate with it like with some function